Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp

```diff
@@ -476,55 +480,78 @@ void pfPatcherWorker::Run()
 
 void pfPatcherWorker::ProcessFile()
 {
+    plFileName clientDir = plFileSystem::GetCurrentAppPath().StripFileName();
+
     do {
         NetCliFileManifestEntry& entry = fQueuedFiles.front();
+        fQueuedFiles.pop_front();
 
         // eap sucks
-        plFileName clName = ST::string::from_wchar(entry.clientName);
+        plFileName clName = plFileName(ST::string::from_wchar(entry.clientName)).Normalize();
         ST::string dlName = ST::string::from_wchar(entry.downloadName);
 
+        // Only operate on files if they are contained within the client directory...
+        plFileName clAbsPath = clName.AbsolutePath();
+        if (!clAbsPath.AsString().starts_with(clientDir.AsString())) {
```

Comment:
That's a good point regarding the Python, and it did cross my mind.  I would definitely like to get @zrax and @dpogue to weigh in on this concept before merging it.